src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -295,6 +300,8 @@ private[pulsar] case class PulsarMetadataReader(
       case None =>
         throw new RuntimeException("Failed to get topics from configurations")
     }
+
+    topicQuery()
```

Comment:
`waitForTopicIfNeeded()`

Reply:
> did you try to run this feature locally?

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/1761698/216383316-1b334d3b-4372-4469-a441-bd17e026ab7e.png">

I test the logic by using worksheet. But I don't know how to write the unit test.